feat: add speakeasy run command

TristanSpeakEasy opened the pull request.

chase-crumbaugh commented:
A couple things I found in testing. It's possible some of these aren't related to your change specifically, I'm not sure.

- When using a remote url as a source, I get this error: `ERROR	failed to get schema contents: failed to download OpenAPI schema: failed to download file: Get ".speakeasy/temp/output_qtKuSwTmsp.yaml": unsupported protocol scheme ""`. I tested a few different basic petstore-type sample specs and they all gave that. Notably when I put the remote URL directly in the `source` field for a target, it downloads successfully
- Typescript compilation failed for me. Error was: `npm ERR! Missing script: "check"`. Not sure how to proceed from there
- When I changed my spec source from a local file to a remote one, the local file I had in the repo was deleted

chase-crumbaugh commented on go.mod:
```diff
@@ -155,3 +155,9 @@ require (
 )
+
+// replace github.com/speakeasy-api/sdk-gen-config => github.com/speakeasy-api/sdk-gen-config workflow-files
+replace github.com/speakeasy-api/sdk-gen-config => github.com/speakeasy-api/sdk-gen-config v1.5.2-0.20240109131902-459ea0b170f7
```
Just a reminder not to forget about these

TristanSpeakEasy replied:
done

chase-crumbaugh commented on internal/run/run.go:
```diff
@@ -0,0 +1,310 @@
+		} else if len(wf.Targets) == 0 && len(wf.Sources) == 1 {
+			source = sources[0]
+		} else {
+			// TODO update to use our proper interactive code
```
I'll update this, either in this PR or my own, depending on which gets merged first

chase-crumbaugh commented on internal/run/run.go:
```diff
@@ -0,0 +1,310 @@
+		}
+	}
+
+	if source != "" && source != "all" {
```
There's a bit of weirdness here I think. I added two sources, `s1` and `s2`. I set up a target that referenced `s1`. If I run just `speakeasy run`, it runs `s1` and the target. If i run `speakeasy run -s s2` it just runs s2. If i run `speakeasy run -s s2 -t all` it runs just s1 and the target.

Imo the solution to this might be to not allow -s and -t to be specificified at the same time. If you are generating targets, we will determine what sources need to be generated. `-s` should be a special case to tell us to _only_ generate sources

TristanSpeakEasy replied:
okay have dealt with this you can either specify a source or a target

TristanSpeakEasy commented:
> A couple things I found in testing. It's possible some of these aren't related to your change specifically, I'm not sure.
> 
> * When using a remote url as a source, I get this error: `ERROR	failed to get schema contents: failed to download OpenAPI schema: failed to download file: Get ".speakeasy/temp/output_qtKuSwTmsp.yaml": unsupported protocol scheme ""`. I tested a few different basic petstore-type sample specs and they all gave that. Notably when I put the remote URL directly in the `source` field for a target, it downloads successfully
> * Typescript compilation failed for me. Error was: `npm ERR! Missing script: "check"`. Not sure how to proceed from there
> * When I changed my spec source from a local file to a remote one, the local file I had in the repo was deleted

@chase-crumbaugh can you link me to the spec and workflow file you were using that cause the typescript compilation, and the spec the url that causes the unsupported protocol scheme error?

TristanSpeakEasy commented:
> > A couple things I found in testing. It's possible some of these aren't related to your change specifically, I'm not sure.
> > 
> > * When using a remote url as a source, I get this error: `ERROR	failed to get schema contents: failed to download OpenAPI schema: failed to download file: Get ".speakeasy/temp/output_qtKuSwTmsp.yaml": unsupported protocol scheme ""`. I tested a few different basic petstore-type sample specs and they all gave that. Notably when I put the remote URL directly in the `source` field for a target, it downloads successfully
> > * Typescript compilation failed for me. Error was: `npm ERR! Missing script: "check"`. Not sure how to proceed from there
> > * When I changed my spec source from a local file to a remote one, the local file I had in the repo was deleted
> 
> @chase-crumbaugh can you link me to the spec and workflow file you were using that cause the typescript compilation, and the spec the url that causes the unsupported protocol scheme error?

Okay I have fixed the unsupported protocol scheme error but still need help with some reproduction steps for the typescript issue

TristanSpeakEasy commented:
> * When I changed my spec source from a local file to a remote one, the local file I had in the repo was deleted

@chase-crumbaugh can I get more details on this?

TristanSpeakEasy commented on internal/run/run.go:
```diff
@@ -0,0 +1,310 @@
+		return err
+	}
+
+	wf, _, err := workflow.Load(workingDir)
```
I have added more detail to this error
